crypto.go

```diff
@@ -3,22 +3,51 @@ package e3dbClients
 import (
 	"crypto/rand"
 	"encoding/base64"
+	"errors"
+	"fmt"
 
+	"github.com/google/uuid"
+	"github.com/tozny/e3db-go/v2"
+	"golang.org/x/crypto/nacl/box"
+	"golang.org/x/crypto/nacl/secretbox"
 	sodium "golang.org/x/crypto/nacl/sign"
 )
 
 const (
 	// https://security.stackexchange.com/questions/50878/ecdsa-vs-ecdh-vs-ed25519-vs-curve25519
 	DefaultEncryptionKeyType = "curve25519"
 	DefaultSigningKeyType    = "ed25519"
+	SymmetricKeySize         = 32
+	NonceSize                = 24
 )
 
+// SymmetricKey is used for fast encryption of larger amounts of data
+// also referred to as the 'SecretKey'
+type SymmetricKey *[SymmetricKeySize]byte
+
+// Nonce is a type that represents nonce randomly generated unique value that is used once in encrypting data.
+type Nonce *[NonceSize]byte
+
+// ToznyAuthenticatedClientContext represents the contextual information provided by cyclops to downstream services
+// when a user is successfully authenticated.
+type ToznyAuthenticatedClientContext struct {
+	ClientID       uuid.UUID            `json:"client_id"`
+	AccountID      uuid.UUID            `json:"account_id"`
+	Name           string               `json:"name"`
+	EncryptionKeys PublicEncryptionKeys `json:"encryption_keys"`        // Tozny does not know a user's private encryption key
+	SignatureKeys  PublicSignatureKeys  `json:"signing_keys,omitempty"` // Tozny does not know a user's private signing key
+	ClientType     string               `json:"type"`
+}
+
 // Key wraps material generated using an algorithm/curve for use in cryptographic operations
 type Key struct {
 	Material string
 	Type     string // e.g. Curve25519
 }
 
+// Keys is a map of key type to key material
+type Keys map[string]string
```

Comment:
Should this just be a list of Keys? If not, what about adding types instead of string->string to indicate what those strings are used for?

Reply:
For rolling keys: We'd probably want a `map[KeyId or KeyNumber]Key` anyway. For keys of varying `Type`s, the information is already captured in `Key`

Reply:
Alternatively, if this is just a serialization format, should it be exported at all?

Reply:
This future functionality is more for modes over rolling keys such as NIST vs Sodium mode; hence the map structure. We thought about adding types, but it adds too much overhead in creation/using the type that I think it's not worth it.